httprouter/bearerstdapi/bearerstdapi.go

```diff
@@ -66,6 +104,64 @@ func (b *BearerStandardAPI) ProcessData(req *http.Request) (interface{}, error)
 	}, nil
 }
 
+// RegisterMethod adds a new method under the URL pattern.
+// The pattern can contain variable names (starting with #), such as /send/#name/hello
+// The pattern can also contain wildcard at the end of the path, such as /send/#name/*
```

Comment:
Did you try reusing chi? It already supports these kind of variables and path matches out of the box; see https://github.com/go-chi/chi. And it's the HTTP URL router we've used elsewhere already.

I know this is not a lot of code, but I'd rather not reinvent the wheel. Some of the code you wrote down here is hard to follow without any docs, for example, and it's hard to say how many edge case bugs it has.

Reply:
Wait, we are *already* using chi under the hood, just two levels down, through httprouter. Why not just use the chi route path syntax directly? A new syntax on top that gets translated (e.g. `/#foo` to `/{foo}`) seems really confusing. It definitely confused me to think we had implemented our own router from scratch.

Reply:
I was waiting for this comment, haha. I got the same thinking as you did while I was implementing the custom URL variable parser.

I need to think about how to use the go-chi Mux for this while preserving the abstraction layer of the httpRouter clean (so we can switch from go-chi to something else). Do you have a proposal?

Btw, the /{foo}/ is just added cause go-chi does not allow to use /*/ if its not the end of the path (just a workaround, but I'm not using the {foo} value obtained by go-chi.

Reply:
I see, I wasn't aware you were adding an abstraction layer on purpose. I can't help but think it's overkill here. Abstraction layers for transports, encryption mechanisms, or even message encodings make sense - it's likely we'll support at least a couple of implementations for each sooner than later. I personally don't think that will be the case with chi, as it's a pretty complete and performant http router. I don't see why we need to pay the price of the abstraction.

Reply:
> Btw, the /{foo}/ is just added cause go-chi does not allow to use /*/ if its not the end of the path (just a workaround, but I'm not using the {foo} value obtained by go-chi.

that seems fine - I'd perhaps consider `{discard}` just to clarify we discard the value

Reply:
> I see, I wasn't aware you were adding an abstraction layer on purpose. I can't help but think it's overkill here. Abstraction layers for transports, encryption mechanisms, or even message encodings make sense - it's likely we'll support at least a couple of implementations for each sooner than later. I personally don't think that will be the case with chi, as it's a pretty complete and performant http router. I don't see why we need to pay the price of the abstraction.

I think the abstraction layer makes sense because now we have to very different API approaches:

- signed json RPC: the abstraction layer handles the JSON message (which has a specific, predefined format), check signature and resources access, handle error messaging, etc (close to what multirpc does)
- bearer URL API: the abstraction layer handles the bearer Auth, error messages (which have different format) and also the URL variable parsing (which is not required by the JSON RPC)

The httpRouter makes makes that these two implementations (and possible future ones) could coexist within the same http server (go-chi) in a unified and standarized way.

I really think it is quite consistent, the only question is if we could reuse the go-chi variable parsing instead of our own. But as you already said, its just a few lines of code, so not that painful. I'll study changing it (in order to not reinvent the wheel) but for now the current implementation works. We can change it in the future without a huge refactor.

Reply:
> > Btw, the /{foo}/ is just added cause go-chi does not allow to use /*/ if its not the end of the path (just a workaround, but I'm not using the {foo} value obtained by go-chi.
> 
> that seems fine - I'd perhaps consider `{discard}` just to clarify we discard the value

does not work cause var names cannot be repeated in go-chi. So I'd need to use {discard1} {discard2} which is even more ugly (I would say).

Reply:
incremental changes sound good to me!

Reply:
I found a quite elegant and clean solution (easy to apply) that allows using go-chi url parser instead of our own: https://github.com/vocdoni/vocdoni-node/pull/359/files#diff-9eb7c1f117ca67d37752f846cc98bfeaf31a980c9eedf181d547fae87a95d8faR30

Reply:
So now use the same go-chi syntax: `/entities/{entity}/processes/{status}`, handlers can get the vars using the HTTP context (https://github.com/vocdoni/vocdoni-node/pull/359/files#diff-2cff19f634c3d726591bf8ffc4119faac8b72b3457b77bb4a18bdbf0b2d814a9R48) and we still preserve the abstraction layer (changing go-chi by another http-server would only require changing the httpRouter layer).

Reply:
great!